apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -338,7 +347,11 @@ public static SpannerDbType FromClrType(System.Type type)
             {
                 return Timestamp;
             }
-            if (type == typeof(float) || type == typeof(double) || type == typeof(decimal))
+            if (type == typeof(float))
+            {
+                return Float32;
+            }
+            if (type == typeof(double) || type == typeof(decimal))
```

Comment:
If we're making breaking changes anyway, maybe we should be using SpannerNumeric for decimal? (In a separate PR no doubt, but let's discuss)

Reply:
Yes, we've agreed to do this, I'll send a PR later.